cylc/flow/etc/job.sh

```diff
@@ -193,6 +216,14 @@ cylc__job_finish_err() {
         wait "${CYLC_TASK_MESSAGE_STARTED_PID}" 2>'/dev/null' || true
     fi
     cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" "$@" || true
```

Comment:
This could possibly be run as a background process similarly to CYLC_TASK_MESSAGE_STARTED_PID to let the rest of the trap code carry on without waiting on network communication.